### DISCUSSION TITLE
Missing {subject}

### DISCUSSION THREAD

**Discussion by LZhangCWRU** [HALFpipe/HALFpipe#548] [Q&A] [answered]
I am using Docker version halfpipe on my Mac (M1 chip) as below, and it shows warning message:

>docker run --interactive --tty --volume /:/ext halfpipe/halfpipe
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested

Then: It shows missing {subject} as attached with BIDS format NO. My data file name as attached 2.

<img width="741" alt="halfpipe" src="https://github.com/HALFpipe/HALFpipe/assets/125577571/50e08d3a-359d-4eab-82f4-33b7c6afa700">
<img width="265" alt="data_structure" src="https://github.com/HALFpipe/HALFpipe/assets/125577571/30f0eb4b-5890-4637-b651-ea30cd31912f">

If I chose BIDS format Yes, it shows below under my sub01 folder:
Found 0 T1-weighted image files
Found 0 BOLD image files

<img width="364" alt="halfpipe2" src="https://github.com/HALFpipe/HALFpipe/assets/125577571/5703c1ff-aca5-451c-867c-5f86a148eb1c">

Do you know the problem could be related my Mac with M1 chip or the halfpipe version?

Thanks

**Comment by HippocampusGirl** [comment, marked as answer]
Hi @LZhangCWRU,

- The warning message can be ignored I think.

- Could you try putting `{subject}` in place of the subject IDs? That means the path from the first screenshot
```
/Users/lzhang/Downloads/TestHC/sub01/sub01_T1.nii.gz
```
would become
```
/Users/lzhang/Downloads/TestHC/sub{subject}/sub{subject}_T1.nii.gz
```
This way, the program can extract the subject IDs from the filenames.

- I would also recommend creating a new folder for your working directory, because having the raw data and the outputs in the same directory can sometimes lead to problems down the line.

Hope this helps!

**Reply by LZhangCWRU** [reply]
Thanks HippocampusGirl
Now it has new error with "string index out of range" as attached, I also attach error file.
<img width="932" alt="error" src="https://github.com/HALFpipe/HALFpipe/assets/125577571/56ee9e82-5b1a-45c0-b4bc-187d414cb37a">


[spec.json.txt](https://github.com/HALFpipe/HALFpipe/files/11455523/spec.json.txt)

[log.txt](https://github.com/HALFpipe/HALFpipe/files/11455524/log.txt)


Would you check what the problem could be?

Thanks

**Comment by LZhangCWRU** [comment]
Lea:
Thanks for helping me with the missing subject problem.

Now it has a new error with "string index out of range" as attached, I also
attach an error file below.

I ran halfpipe both on my Mac (using Docker) and our HPC with
(singularity/3.8.1). Both show this string index error. I try to do the
preprocessing step without the first-level feature.


[image: error]
<https://user-images.githubusercontent.com/125577571/237786056-56ee9e82-5b1a-45c0-b4bc-187d414cb37a.png>

spec.json.txt
<https://github.com/HALFpipe/HALFpipe/files/11455523/spec.json.txt>

log.txt <https://github.com/HALFpipe/HALFpipe/files/11455524/log.txt>

Would you check what the problem could be?
Thanks
Best

**Reply by HippocampusGirl** [reply]
Hi @LZhangCWRU, 
I'm out of the office this week but will have a look next week! 

Best,
Lea
